### PR TITLE
Ensure `cargo check` works for the workspace without flags

### PR DESCRIPTION
**Pull request by mraerino**
This makes sure we check the platform specific crates using the correct architecture.

See https://github.com/rust-lang/cargo/pull/9030 for the unstable cargo feature.

### CODE REVIEW

File: xtask/src/main.rs [hunk 1]
```diff
@@ -397,7 +397,7 @@ fn test(build_params: &BuildParams) -> Result<()> {
     cmd.current_dir(workspace());
     cmd.arg("test");
     cmd.arg("--workspace");
-    cmd.arg("--config").arg("build.rustflags='--cfg platform=\"virt\"'");
+    cmd.arg("--target").arg("x86_64-unknown-linux-gnu");
```
Comment (gmacd) [inline xtask/src/main.rs:400]: why is this change necessary?

Comment (mraerino) [reply xtask/src/main.rs:400]: because the tests rely on the stdlib which the default targets that are now set in `Cargo.toml` are not providing. sadly i've found the tests generally only seem to work on `x86_64-unknown-linux-gnu` (sad macbook noises)

i removed the `--config` arg since it is not required (this fallback is already part of the `cfg` macro in the code

**Review by orangecms — APPROVED** [review]
LGTM

**Review by gmacd — APPROVED** [review]
